Snowflake.Data.Tests/ConcatenatedStreamTest.cs

```diff
@@ -0,0 +1,159 @@
+﻿/*
+ * Copyright (c) 2023 Snowflake Computing Inc. All rights reserved.
+ */
+
+namespace Snowflake.Data.Tests
+{
+    using NUnit.Framework;
+    using Snowflake.Data.Core;
+    using System;
+    using System.IO;
+    using System.Text;
+
+    [TestFixture]
+    class ConcatenatedStreamTest
+    {
+        Stream concatStream;
+
+        [SetUp]
+        public void BeforeTest()
+        {
+            string data = "12345678";
+            byte[] bytes = Encoding.UTF8.GetBytes(data);
+            Stream stream = new MemoryStream(bytes);
+
+            Stream openBracket = new MemoryStream(Encoding.UTF8.GetBytes(data));
+            Stream closeBracket = new MemoryStream(Encoding.UTF8.GetBytes(data));
+
+            concatStream = new ConcatenatedStream(new Stream[3] { openBracket, stream, closeBracket });
+        }
+
+        [Test]
+        [Ignore("ConcatenatedStreamTest")]
+        public void ConcatenatedStreamTestDone()
+        {
+            // Do nothing;
+        }
+
+        [Test]
+        public void TestCanRead()
+        {
+            Assert.IsTrue(concatStream.CanRead);
+        }
+
+        [Test]
+        public void TestCanSeek()
+        {
+            Assert.IsFalse(concatStream.CanSeek);
+        }
+
+        [Test]
+        public void TestCanWrite()
+        {
+            Assert.IsFalse(concatStream.CanWrite);
+        }
+
+        [Test]
+        public void TestFlush()
+        {
+            try
+            {
+                concatStream.Flush();
```

Comment:
if we expect this line to throw exception then just after it we should add always failing assertion because if that line won't fail then the test will be green

Reply:
in other tests I saw usage of `Assert.Throws<SpecificExceptionClass>` - maybe this method can be used instead of explicit try/catch?

Reply:
Ok

Reply:
instead of try..catch it rather should be something like:
Assert.Throws< NotImplementedException>(() => concatStream.Flush());